SQLAssignment.sql.sql

```diff
@@ -0,0 +1,1043 @@
+| 2004 | Grass    | Berlin  |    300 | 1002 |
+| 2008 | Cisneros | SanJose |    300 | 1007 |
++------+----------+---------+--------+------+
+24) Count the number of salespeople currently having orders in the orders table.
+mysql> SELECT  COUNT(DISTINCT SNUM) AS No_Of_Salespeople FROM Orders;
++-------------------+
+| No_Of_Salespeople |
++-------------------+
+|                 5 |
++-------------------+
+25) Write a query that produces all customers serviced by salespeople with a commission above 12%. Output the customer’s name,
+salesperson’s name and the salesperson’s rate of commission.
+mysql> SELECT Customers.CNAME,Salespeople.SNAME,Customers.RATING,Salespeople.COMM 
+       FROM Customers,Salespeople 
+	   WHERE Customers.SNUM=Salespeople.SNUM AND Salespeople.COMM>12;
++----------+--------+--------+------+
+| CNAME    | SNAME  | RATING | COMM |
++----------+--------+--------+------+
+| Liu      | Serres |    200 |   13 |
+| Grass    | Serres |    300 |   13 |
+| Cisneros | Rifkin |    300 |   15 |
++----------+--------+--------+------+
+26) Find salespeople who have multiple customers.
+mysql> SELECT SNUM, COUNT(SNUM) AS Total_Customers FROM Customers GROUP BY SNUM HAVING COUNT(SNUM)>1;
++------+-----------------+
+| SNUM | Total_Customers |
++------+-----------------+
+| 1001 |               2 |
+| 1002 |               2 |
++------+-----------------+
+27) Find salespeople with customers located in their own cities.
+mysql> SELECT s.SNAME,c.CNAME,s.CITY FROM Salespeople s JOIN  Customers c ON s.CITY = c.CITY ORDER BY  c.city;
++--------+----------+---------+
+| SNAME  | CNAME    | CITY    |
++--------+----------+---------+
+| Fran   | Hoffman  | London  |
+| Motika | Hoffman  | London  |
+| Peel   | Hoffman  | London  |
+| Fran   | Clemens  | London  |
+| Motika | Clemens  | London  |
+| Peel   | Clemens  | London  |
+| Serres | Liu      | SanJose |
+| Serres | Cisneros | SanJose |
++--------+----------+---------+
+mysql> SELECT s.SNAME,c.CNAME,s.CITY FROM Salespeople s,Customers c WHERE s.SNUM = c.SNUM AND s.CITY = c.CITY;
++--------+---------+---------+
+| SNAME  | CNAME   | CITY    |
++--------+---------+---------+
+| Peel   | Hoffman | London  |
+| Serres | Liu     | SanJose |
+| Peel   | Clemens | London  |
++--------+---------+---------+
+28) Find all salespeople whose name starts with ‘P’ and fourth character is ‘I’.
+mysql> SELECT SNAME FROM Salespeople  WHERE   SNAME  LIKE   'p__l%' ;
++-------+
+| SNAME |
++-------+
+| Peel  |
++-------+
+29) Write a query that uses a subquery to obtain all orders for the customer named ‘Cisneros’. Assume you do not know his customer number.
+
+mysql> SELECT ONUM FROM Orders WHERE SNUM =(SELECT SNUM FROM Customers WHERE CNAME = 'Cisneros');
++------+
+| ONUM |
++------+
+| 3001 |
+| 3006 |
++------+
+30) Find the largest orders for Serres and Rifkin.
+mysql> SELECT  SNUM,MAX(AMT) 
+       FROM Orders GROUP BY SNUM  HAVING  SNUM IN (SELECT SNUM FROM Salespeople  
```

Comment:
Display the name of the salesperson in the query so it is easy to identify the respective orders by Serres or Rifkin.